src/node/hiding.rs

```diff
@@ -0,0 +1,245 @@
+use crate::node::{
+    CoreConstructible, DisconnectConstructible, JetConstructible, WitnessConstructible,
+};
+use crate::types::{Context, Error};
+use crate::{Cmr, FailEntropy, HasCmr, Word};
+
+/// Wrapper that allows a node to be "hidden" during program construction.
+///
+/// ## Program construction
+///
+/// When a program is constructed in post-order,
+/// the parent node is created based on its children.
+/// We use this fact to introduce special branching logic:
+///
+/// 1. A `case` node with a left "hidden" child and a right non-hidden child becomes `assertr`.
+/// 2. A `case` node with a left non-hidden child and a right "hidden" child becomes `assertl`.
+/// 3. Otherwise, any node with "hidden" children becomes itself "hidden" with an updated CMR.
+/// 4. Any node with non-hidden children remains unchanged.
+///
+/// The program can be extracted from the wrapper when construction is finished.
+/// The program is invalid if the root node is "hidden".
+///
+/// ## Wrapping
+///
+/// A node can be wrapped via [`Hidden::from`] to add hiding support.
+/// A wrapped node can be converted into a "hidden" node via [`Hidden::hide`].
+/// Finally, a "hidden" node can be manually created via [`Hidden::hidden`].
```

Comment:
In 8bc68ac1663d0c47bb745a87df168e54a7113be4:

In several places in these docs you say `Hidden` but the type is called `Hiding`.

Reply:
Good catch. Thanks for pointing out the typo.